libsql-server/src/connection/program.rs

```diff
@@ -105,7 +106,7 @@ impl<'a, B, F, S> Vm<'a, B, F, S>
 where
     B: QueryResultBuilder,
     F: Fn(&StmtKind) -> (bool, Option<String>),
-    S: Fn(String, &rusqlite::Statement, Duration),
+    S: Fn(String, u64, u64, u64, Duration),
```

Comment:
Why is this change needed? It seems that the whole PR would be much simpler if we kept accepting Statement here, no?

Reply:
I did this change because it made the most sense, we only want to request these stats once and then pass them to the callback and to the add_stats of the build. In addition, we want these values before we hit the callback. This simplification made the most sense to me rather than leaking a detail like this.

Reply:
I have nothing against this change but it is completely irrelevant to the goal of this PR. It's better to do it in a separate PR as a follow up. Otherwise I needed to spend 50% of time reviewing non-essential change. No need to do anything now but in the future having independent changes in independent PRs or at least commits is a desirable thing.